examples/image_list/test/load_test.dart

```diff
@@ -0,0 +1,553 @@
+import 'dart:async';
+import 'dart:io';
+import 'dart:typed_data';
+
+import 'package:flutter/cupertino.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:image_list/main.dart';
+import 'package:mockito/mockito.dart';
+
+class MockHttpClient extends Mock implements HttpClient {}
+class MockHttpClientRequest extends Mock implements HttpClientRequest {}
+class MockHttpClientResponse extends Mock implements HttpClientResponse {}
+
+const List<int> kSamplePng = <int>[
```

Comment:
Is this images/coast.jpg? Why not read the file directly?

Reply:
Reading file directly (as I attempted at https://github.com/flutter/flutter/pull/41137/commits/cd413306a639269dac5ae664a7f653dcc6c9531b) doesn't work when bytes are needed on a different isolate where image loading is happening.

Reply:
I thought the "loading images on a separate isolate" PR was reverted?

Reply:
Yes, it was reverted, but I am hoping to reland it soon.

Reply:
This seems like another inconvenience of "loading images on a separate isolate" then. Having this random binary data in a test doesn't really seem ideal...

Reply:
Also, why doesn't this work across isolates?

Reply:
Looking at load_test.dart in https://github.com/flutter/flutter/pull/41137/commits/84dc2d50821c561b944083dd424bccc1396acc3a - `bytes` is a top-level field that is populated in `setUpAll()`, but that will be `null` in a image-loading isolate because that is different isolate.

Reply:
Why can the image not be read from disk inside the `sampleHttpProvider` method?

Reply:
To read bytes from the assets I was using `rootBundle`(in that 84dc2d5), which was another global not properly initialized on dedicated isolate.

Reply:
Would it work when read directly from disk bypassing the rootBundle? That seems a little cleaner than having this byte list?

Reply:
Is there a reliable way of reading file from disk directly?
Current directory seems to be different depending on whether you run tests from IJ or from command line.

Reply:
I believe in other tests we use https://master-api.flutter.dev/flutter/dart-io/Platform/script.html and address the file relative to that path.

Reply:
Yeah, that is what I mean - when you run `flutter test` from command line, `Platform.script` is `$FH/flutter/examples/image_list/test/main.dart`, when you run flutter tests from IJ, it is `$FH/flutter/examples/image_list/main.dart`, the difference being `test` folder